File: src/MoManI.Ui/scripts/models/scenario.ts
import uuid = require('node-uuid');

export class Scenario {
    id: string;
    modelId: string;
    name: string;
    description: string;
    revision: number;

    constructor(scenario: IScenario) {
        this.id = scenario.id;
        this.modelId = scenario.modelId;
        this.name = scenario.name;
        this.description = scenario.description;
        this.revision = scenario.revision;
    }

    serialize: () => IScenario = () => {
        return {
            id: this.id,
            modelId: this.modelId,
            name: this.name,
            description: this.description,
            revision: this.revision,
            hasResults: null,
        }
    }
}

File: src/MoManI.Api/Models/Scenario.cs
using System;

namespace MoManI.Api.Models
{
    public class Scenario
    {
        public Guid Id { get; set; }
        public Guid ModelId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Revision { get; set; }

        public Scenario Clone(int revision)
        {
            return new Scenario
            {
                Id = Guid.NewGuid(),
                ModelId = ModelId,
                Name = Name,
                Description = Description,
                Revision = revision,
            };
        }
    }
}
